apps/web/pages/api/integrations/[...args].ts

```diff
@@ -29,21 +31,21 @@ const defaultIntegrationAddHandler = async ({
     const alreadyInstalled = await prisma.credential.findFirst({
       where: {
         appId: slug,
-        userId: user.id,
+        ...(teamId ? { AND: [{ userId: user.id }, { teamId }] } : { userId: user.id }),
       },
     });
     if (alreadyInstalled) {
       throw new Error("App is already installed");
     }
   }
-  await createCredential({ user: user, appType, slug });
+  await createCredential({ user: user, appType, slug, teamId });
 };
 
 const handler = async (req: NextApiRequest, res: NextApiResponse) => {
   // Check that user is authenticated
   req.session = await getServerSession({ req, res });
 
-  const { args } = req.query;
+  const { args, teamId } = req.query;
```

Comment:
Shall we parse this with zod? Makes sure we dont have to cast as number below either (L:71)